### PR TITLE
Add nearest mode to resize_images

### PR DESCRIPTION
**Pull request by yuyu2172**
merge after https://github.com/chainer/chainer/pull/7429

Current implementation of `resize_images` only supports bilinear interpolation.
This PR adds `nearest` interpolation.

### CODE REVIEW

File: chainer/functions/array/resize_images.py [hunk 1]
```diff
@@ -264,10 +296,11 @@ def forward(self, inputs):
         return gx,
 
     def backward(self, indexes, grad_outputs):
-        return ResizeImages((self.out_H, self.out_W)).apply(grad_outputs)
+        return ResizeImages(
+            (self.out_H, self.out_W), self.mode, self.align_corners).apply(grad_outputs)
```
Comment (emcastillo) [inline chainer/functions/array/resize_images.py]: CI Fails on this line: E501 line too long (88 > 79 characters)

File: chainer/functions/array/resize_images.py [hunk 2]
```diff
@@ -169,11 +171,44 @@ def interpolate_grad_bilinear_gpu(gy, v, u, vw, uw, H, W):
     return gx.reshape((B, C, H, W))
 
 
+def compute_indices_and_weights(out_size, in_size, mode, align_corners, xp):
```
Comment (emcastillo) [inline chainer/functions/array/resize_images.py]: Could align_corners and mode be unified in a single parameter?

mode = {'bilinear','bilinear_align','nearest'}

Comment (yuyu2172) [reply chainer/functions/array/resize_images.py]: I am ok with either implementation.
BTW, `align_corners` is quite common option in various DL frameworks. I thought it was easier for users if Chainer's implementation is consistent with others. Is that OK?

Comment (emcastillo) [reply chainer/functions/array/resize_images.py]: Then lets keep that one!
Thanks for the clarification!

**Comment by emcastillo** [comment]
Jenkins, test this please

**Comment by emcastillo** [comment]
Jenkins, test this please

**Comment by emcastillo** [comment]
Jenkins error is not related

**Pull request merged**